vsketch/vsketch.py

```diff
@@ -1832,3 +1833,123 @@ def easing(
+
+    def textMode(self, mode: str) -> None:
+        """Controls how text is laid out.
+
+        There are two options for the text mode:
+            * "transform", where the font itself is subject to the current transform
+              matrix, including translation, scaling, rotation, skewing and flipping.
+            * "label", where the text is only scaled and translated using the current
+              transform matrix, but is not rotated, skewed or flipped.
+
+        .. seealso::
+
+            * :func:`text`
+
+        Args:
+            mode (``"transform"``, or ``"label"``): text layout mode to use.
+        """
+
+        if mode in ["label", "transform"]:
+            self._text_mode = mode
+        else:
+            raise ValueError("mode must be one of 'label', 'transform'")
+
+    def text(
+        self,
+        text: str,
+        x: float = 0.0,
+        y: float = 0.0,
+        *,
+        width: Optional[float] = None,
+        font: str = "futural",
+        size: float = 1.0,
+        mode: Optional[str] = None,
+        align: str = "left",
+        line_spacing: float = 1.0,
+        justify: bool = False,
+    ) -> None:
+        """Add text to your sketch!
+
+        This can add either lines or blocks of text to your sketch. The default is
+        to add a line of text, but if `width` is specified, then a block of text
+        will be created that width.
+
+        The fonts available are the same as those available from vpype.
+
+        A ``size`` of 1.0 means the maximum font height will be as long as a line 1.0
+        long (given current transform).
+
+        The ``mode`` here is special. There are two options for the mode:
+
+            * "transform", where the font itself is subject to the current transform
+              matrix, including translation, scaling, rotation, skewing and flipping.
+            * "label", where the text is only moved and scaled using the current
+              transform matrix, but is not rotated, skewed or flipped.
+
+        The text mode can also be set using :func:`textMode`.
+
+        ``align`` controls the text alignment, and can be "left", "right" or "center".
+
+        ``line_spacing`` and ``justify`` are only used when ``width`` is specified, and they
+        allow you to control the line spacing and justification of your block of text.
+
+        .. seealso::
+
+            * :func:`textMode`
+
+        Args:
+            text: text to add to the sketch
+            x, y: the coordinates of the anchor point for the text.
+                The y coordinate gives the centre line of the first line of text.
+                The x coordinate gives either the leftmost, rightmost, or centre
+                point of the first line of text.
+            width: (optional) if given, wraps the text to this width and creates
+                a text block, not a text line.
+            size: the height of a line of text, in document units.
+            font: a vpype font
```

Comment:
I would like to say you should reference `vpype.FONT_NAMES` in vpype's doc, but for the life of me I havent managed to have module attribute show up in vpype's doc. You could instead link to [here](https://github.com/abey79/vpype/tree/master/vpype/fonts), which can serve as poor-man's list.

Reply:
https://docs.readthedocs.io/en/stable/guides/intersphinx.html is something you could do in future. For the moment, I'll link to what you gave me.

Reply:
Yeah I have that setup already. The issue I have is for FONT_NAMES to actually show up in vpype's documentation so that we may link to it from vsketch doc.